cmd/boulder-observer/README.md

```diff
@@ -0,0 +1,213 @@
+I152530 boulder-observer prOnrw8 kind=[DNS] success=[false] duration=[0.000242] name=[[2606:4700:4700::1111]:53-tcp-A-google.com-recurse]
+I152530 boulder-observer 6uXugQw kind=[DNS] success=[true] duration=[0.022962] name=[1.1.1.1:53-udp-A-google.com-recurse]
+I152530 boulder-observer to7h-wo kind=[DNS] success=[true] duration=[0.029860] name=[owen.ns.cloudflare.com:53-udp-A-letsencrypt.org-no-recurse]
+I152530 boulder-observer ovDorAY kind=[DNS] success=[true] duration=[0.033820] name=[owen.ns.cloudflare.com:53-tcp-A-letsencrypt.org-no-recurse]
+...
+```
+
+## Configuration
+
+Configuration is provided via a YAML file.
+
+### Root
+
+#### Schema
+
+`debugaddr`: The Prometheus scrape port prefixed with a single colon
+(e.g. `:8040`).
+
+`syslog`: Map of log levels, see schema below.
+
+- `stdoutlevel`: Log level for stdout, see legend below.
+- `sysloglevel`:Log level for stdout, see legend below.
+
+`0`: *EMERG* `1`: *ALERT* `2`: *CRIT* `3`: *ERR* `4`: *WARN* `5`:
+*NOTICE* `6`: *INFO* `7`: *DEBUG*
+
+`monitors`: List of monitors, see [monitors](#monitors) for schema.
+
+#### Example
+
+```yaml
+debugaddr: :8040
+syslog:
+  stdoutlevel: 6
+  sysloglevel: 6
+monitors:
+  -
+    ...
+```
+
+### Monitors
+
+#### Schema
+
+`period`: Interval between probing attempts (e.g. `1s` `1m` `1h`).
+
+`kind`: Kind of prober to use, see [probers](#probers) for schema.
+
+`settings`: Map of prober settings, see [probers](#probers) for schema.
+
+#### Example
+
+```yaml
+monitors:
+  - 
+    period: 5s
+    kind: DNS
+    settings:
+        ...
+```
+
+### Probers
+
+#### DNS
+
+##### Schema
+
+`protocol`: Protocol to use, options are: `udp` or `tcp`.
+
+`server`: Hostname, IPv4 address, or IPv6 address surrounded with
+brackets + port of the DNS server to send the query to (e.g.
+`example.com:53`, `1.1.1.1:53`, or `[2606:4700:4700::1111]:53`).
+
+`recurse`: Bool indicating if recursive resolution is desired.
+
+`query_name`: Name to query (e.g. `example.com`).
+
+`query_type`: Record type to query, options are: `A`, `AAAA`, `TXT`, or
+`CAA`.
+
+##### Example
+
+```yaml
+monitors:
+  - 
+    period: 5s
+    kind: DNS
+    settings:
+      protocol: tcp
+      server: [2606:4700:4700::1111]:53
+      recurse: false
+      query_name: letsencrypt.org
+      query_type: A
+```
+
+#### HTTP
+
+##### Schema
+
+`url`: Scheme + Hostname to send a request to (e.g.
+`https://example.com`).
+
+`rcodes`: List of expected HTTP response codes.
+
+##### Example
+
+```yaml
+monitors:
+  - 
+    period: 2s
+    kind: HTTP
+    settings: 
+      url: http://letsencrypt.org/FOO
+      rcodes: [200, 404]
+```
+
+## Metrics
+
+Observer provides the following metrics.
+
+### obs_monitors
+
+Count of configured monitors.
+
+**Labels:**
+
+`name`: Name of the monitor.
+
+`kind`: Kind of prober the monitor is configured to use.
+
+`valid`: Bool indicating whether the result of the probe attempt.
```

Comment:
whether the result of the probe attempt what?

Also I thought this one became `false` when the prober was unable to bet set up, not when the probe attempt failed.